gen/gen.go

```diff
@@ -44,10 +44,11 @@ func init() {
 	}
 }
 
-func Gen(jFile string, j interface{}, t *template.Template) error {
+func Gen(exercise string, j interface{}, t *template.Template) error {
 	if dirMetadata == "" {
 		return errors.New("unable to determine current path")
 	}
+	jFile := filepath.Join("exercises", exercise, "canonical-data.json")
```

Comment:
ok, super, I'll leave it as is, as you say it's not hard to understand nor is it repeated. I do like the idea of a `jsonPath` function but it also feels like maybe one layer of abstraction too many.